node/tests/swamp/swamp.go

```diff
@@ -178,76 +178,54 @@ func (s *Swamp) getTrustedHash(ctx context.Context) (string, error) {
 }
 
 // NewBridgeNode creates a new instance of a BridgeNode providing a default config
-// and a mockstore to the NewBridgeNodeWithStore method
+// and a mockstore to the NewNodeWithStore method
 func (s *Swamp) NewBridgeNode(options ...node.Option) *node.Node {
 	cfg := node.DefaultConfig(node.Bridge)
 	store := node.MockStore(s.t, cfg)
 
-	return s.NewBridgeNodeWithStore(store, options...)
+	return s.NewNodeWithStore(node.Bridge, store, options...)
 }
 
 // NewFullNode creates a new instance of a FullNode providing a default config
-// and a mockstore to the NewFullNodeWithStore method
+// and a mockstore to the NewNodeWithStore method
 func (s *Swamp) NewFullNode(options ...node.Option) *node.Node {
 	cfg := node.DefaultConfig(node.Full)
 	store := node.MockStore(s.t, cfg)
 
-	return s.NewFullNodeWithStore(store, options...)
+	return s.NewNodeWithStore(node.Full, store, options...)
 }
 
 // NewLightNode creates a new instance of a LightNode providing a default config
-// and a mockstore to the NewLightNodeWithStore method
+// and a mockstore to the NewNodeWithStore method
 func (s *Swamp) NewLightNode(options ...node.Option) *node.Node {
 	cfg := node.DefaultConfig(node.Light)
 	store := node.MockStore(s.t, cfg)
 
-	return s.NewLightNodeWithStore(store, options...)
+	return s.NewNodeWithStore(node.Light, store, options...)
 }
 
-// NewBridgeNodeWithStore creates a new instance of BridgeNodes with predefined Store.
-// Afterwards, the instance is stored in the swamp's BridgeNodes slice.
-func (s *Swamp) NewBridgeNodeWithStore(store node.Store, options ...node.Option) *node.Node {
-	ks, err := store.Keystore()
-	require.NoError(s.t, err)
-
-	// TODO(@Bidon15): If for some reason, we receive one of existing options
-	// like <core, host, hash> from the test case, we need to check them and not use
-	// default that are set here
-	options = append(options,
-		node.WithCoreClient(s.CoreClient),
-		node.WithHost(s.createPeer(ks)),
-		node.WithTrustedHash(s.trustedHash),
-	)
-
-	node, err := node.New(node.Bridge, store, options...)
-	require.NoError(s.t, err)
-	s.BridgeNodes = append(s.BridgeNodes, node)
-	return node
-}
+func (s *Swamp) NewNodeWithStore(t node.Type, store node.Store, options ...node.Option) *node.Node {
```

Comment:
Maybe add a godoc comment for `NewNodeWithStore`.

Reply:
I'll add the godoc in a separate PR with some nits changes to not ask for another round of reviews 🙏